src/module_library/leaf_energy_balance.cpp

```diff
@@ -0,0 +1,223 @@
+    const double p_w_sat_air = saturation_vapor_pressure(air_temperature);               // Pa
+    const double rho_ta = dry_air_density(air_temperature, air_pressure);                // kg / m^3
+    const double s = TempToSFS(air_temperature);                                         // kg / m^3 / K
+
+    // Get the pyschrometric parameter
+    const double gamma = rho_ta * c_p / lambda;  // kg / m^3 / K
+
+    // Get vapor density in the ambient air.
+    const double p_w_air = p_w_sat_air * relative_humidity;  // Pa
+
+    const double rho_w_air =
+        vapor_density_from_pressure(rho_ta, air_pressure, p_w_air);  // kg / m^3
+
+    // Get vapor density deficit
+    const double rho_w_sat =
+        vapor_density_from_pressure(rho_ta, air_pressure, p_w_sat_air);  // kg / m^3
+
+    const double Delta_rho = rho_w_sat - rho_w_air;  // kg / m^3
+
+    // Get total absorbed light energy (longwave and shortwave)
+    const double J_a = absorbed_shortwave_energy + absorbed_longwave_energy;  // J / m^2 / s
+
+    // Make an initial guess for the leaf temperature
+    double leaf_temperature{air_temperature + 0.1};  // degrees C
+
+    // Initialize other loop variables; their values will be set during the loop
+    double gbw_leaf{};   // m / s
+    double gbw{};        // m / s
+    double gsw{};        // m / s
+    double gw{};         // m / s
+    double mv_tl{};      // m^3 / mol
+    double Phi_N{};      // J / m^2 / s
+    double pm_bottom{};  // J / m^3 / K
+    double pm_top{};     // J / m^3
+    double R_l{};        // J / m^2 / s
+
+    // Run loop to find the leaf temperature
+    double constexpr max_delta{30};  // degrees C
+    double old_leaf_temperature{};   // degrees C
+    double change_in_tl{};           // degrees C
+    int counter{0};
+
+    do {
+        // Store leaf temperature from previous loop iteration
+        old_leaf_temperature = leaf_temperature;  // degrees C
+
+        // Get longwave energy losses
+        R_l = epsilon_s * physical_constants::stefan_boltzmann *
+              pow(conversion_constants::celsius_to_kelvin + leaf_temperature, 4);  // J / m^2 / s
+
+        // Get the energy available for transpiration and heat loss
+        Phi_N = J_a - R_l;  // J / m^2 / s
+
+        // Get stomatal conductance to water vapor as a mass conductance
+        mv_tl = molar_volume(leaf_temperature, air_pressure);  // m^3 / mol
+        gsw = stomatal_conductance * mv_tl;                    // m / s
+
+        // Get leaf boundary layer conductance to water vapor
+        gbw_leaf = leaf_boundary_layer_conductance_nikolov(
+            air_temperature,
+            leaf_temperature - air_temperature,
+            p_w_air,
+            gsw,
+            leaf_width,
+            wind_speed,
+            air_pressure);  // m / s
+
+        // Get the boundary layer conductance and total conductance to water
+        // vapor
+        gbw = 1.0 / (1.0 / gbw_leaf + 1.0 / gbw_canopy);  // m / s
```

Comment:
Why do the `gbw_leaf` and `gbw_canopy` need to be inverted for addition?

Reply:
This is analogous to resistances and conductances in series for electrical circuits. Resistances in series are summed. Since conductance is the inverse of resistance, the equivalent sort of calculations is the inverse of the sum of inverses.

`r_t = r_1 + r_2`. If `c_1 = 1 / r_1` and `c_2 = 1 / r_2`, then `r_t = 1 / c_1 + 1 / c_2`, and thus `c_t = 1 / r_t = 1 / ( 1 / c_1 + 1 / c_2)`.